holidays/countries/vietnam.py

```diff
@@ -41,53 +42,40 @@ def _populate(self, year):
         name = "International New Year's Day"
         first_date = date(year, JAN, 1)
         self[first_date] = name
-        if self.observed:
-            self[first_date] = name
-            if first_date.weekday() == SAT:
-                self[first_date + rd(days=+2)] = name + " observed"
-            elif first_date.weekday() == SUN:
-                self[first_date + rd(days=+1)] = name + " observed"
+        if self.observed and first_date.weekday() in WEEKEND:
+            self[first_date + rd(weekday=MO)] = name + " observed"
 
         # Lunar New Year
-        name = [
+        names = [
             "Vietnamese New Year",  # index: 0
             "The second day of Tet Holiday",  # index: 1
             "The third day of Tet Holiday",  # index: 2
             "The forth day of Tet Holiday",  # index: 3
             "The fifth day of Tet Holiday",  # index: 4
             "Vietnamese New Year's Eve",  # index: -1
         ]
-        dt = self.get_solar_date(year, 1, 1)
-        new_year_date = date(dt.year, dt.month, dt.day)
-        if self.observed:
-            for i in range(-1, 5, 1):
-                tet_day = new_year_date + rd(days=+i)
-                self[tet_day] = name[i]
+        new_year_date = self.get_solar_date(year, 1, 1)
+        for i in range(-1, 5, 1):
+            tet_day = new_year_date + rd(days=+i)
+            self[tet_day] = names[i]
 
         # Vietnamese Kings' Commemoration Day
         # https://en.wikipedia.org/wiki/H%C3%B9ng_Kings%27_Festival
         if year >= 2007:
-            name = "Hung Kings Commemoration Day"
-            dt = self.get_solar_date(year, 3, 10)
-            king_hung_date = date(dt.year, dt.month, dt.day)
-            self[king_hung_date] = name
-        else:
-            pass
+            king_hung_date = self.get_solar_date(year, 3, 10)
+            self[king_hung_date] = "Hung Kings Commemoration Day"
 
         # Liberation Day/Reunification Day
-        name = "Liberation Day/Reunification Day"
         libration_date = date(year, APR, 30)
```

Comment:
As the date objects being used once only I'd  optimize them with `self[date(year, APR, 30)] = "Liberation Day/Reunification Day"` approach.

Reply:
Done. Thank you!